pdns/rec_channel_rec.cc

```diff
@@ -242,8 +242,22 @@ template<typename T>
 string doWipeCache(T begin, T end)
 {
   int count=0, countNeg=0;
+
+  vector<string> toWipe;
+
   for(T i=begin; i != end; ++i) {
     string canon=toCanonic("", *i);
+
+    if (canon[0] == '.')
+      return "dot found in the first place of " + canon + ". Nothing wiped.\n";
+
+    if (canon.find("..") != string::npos)
+      return "double dot found in the first place of " + canon + ". Nothing wiped.\n";
```

Comment:
The message may be a bit misleading, since the double dot might not be in the first place.